langchain4j-core/src/main/java/dev/langchain4j/model/chat/ChatLanguageModel.java

```diff
@@ -5,21 +5,141 @@
+                response = generate(chatRequest.messages(), chatParameters.toolSpecifications().get(0));
+            } else {
+                response = generate(chatRequest.messages(), chatParameters.toolSpecifications());
+            }
+        }
+
+        return ChatResponse.builder()
+                .aiMessage(response.content())
+                .metadata(ChatResponseMetadata.builder()
+                        .tokenUsage(response.tokenUsage())
+                        .finishReason(response.finishReason())
+                        .build())
+                .build();
+    }
+
+    static void validate(ChatRequest chatRequest, Class<?> modelClass) {
+        ChatParameters parameters = chatRequest.parameters();
+
+        String errorTemplate = "%s is not supported yet by this model provider";
+
+        if (parameters.modelName() != null) {
+            throw new UnsupportedFeatureException(errorTemplate.formatted("'modelName' parameter"));
+        }
+
+        Class<? extends ChatRequest> chatRequestClass = chatRequest.getClass();
+        if (chatRequestClass != ChatRequest.class) {
+            throw new IllegalArgumentException("%s cannot be used together with %s. Please use %s instead.".formatted(
+                    chatRequestClass.getSimpleName(),
+                    modelClass.getSimpleName(),
+                    ChatRequest.class.getSimpleName()
+            ));
+        }
+        if (parameters.temperature() != null) {
+            throw new UnsupportedFeatureException(errorTemplate.formatted("'temperature' parameter"));
+        }
+        if (parameters.topP() != null) {
+            throw new UnsupportedFeatureException(errorTemplate.formatted("'topP' parameter"));
+        }
+        if (parameters.topK() != null) {
+            throw new UnsupportedFeatureException(errorTemplate.formatted("'topK' parameter"));
+        }
+        if (parameters.frequencyPenalty() != null) {
+            throw new UnsupportedFeatureException(errorTemplate.formatted("'frequencyPenalty' parameter"));
+        }
+        if (parameters.presencePenalty() != null) {
+            throw new UnsupportedFeatureException(errorTemplate.formatted("'presencePenalty' parameter"));
+        }
+        if (parameters.maxOutputTokens() != null) {
+            throw new UnsupportedFeatureException(errorTemplate.formatted("'maxOutputTokens' parameter"));
+        }
+        if (parameters.stopSequences() != null) {
+            throw new UnsupportedFeatureException(errorTemplate.formatted("'stopSequences' parameter"));
+        }
+
+        List<ToolSpecification> toolSpecifications = parameters.toolSpecifications();
+        if (parameters.toolChoice() == REQUIRED
+                && !isNullOrEmpty(toolSpecifications) && toolSpecifications.size() > 1) {
+            throw new UnsupportedFeatureException(
+                    "ToolChoice.REQUIRED is currently supported only when there is a single tool");
```

Comment:
Not really, this is just for backwards compatibility, and temporarily. Since `chat()` is calling `generate()` methods by default for a bit (untill integrations implement their own `chat()` version), we need to ensure that one of `generate()` methods can be actually called.